<!-- Launcher for pose controlling Kobuki -->

<launch>
  <include file="$(find kobuki_node)/launch/robot_with_tf.launch"/>

  <!-- remember to change speed limits for the velocity smoother as , when changing for the pose controller -->
  <node pkg="nodelet" type="nodelet" name="pose_controller_velocity_smoother"
        args="load yocs_velocity_smoother/VelocitySmootherNodelet /mobile_base_nodelet_manager">
    <rosparam file="$(find waiterbot_ctrl_nowireless)/param/vel_smoother_pose_controller.yaml" command="load"/>
    <remap from="pose_controller_velocity_smoother/smooth_cmd_vel" to="/mobile_base/commands/velocity"/>
    <remap from="pose_controller_velocity_smoother/robot_cmd_vel" to="/mobile_base/commands/velocity"/>
    <remap from="pose_controller_velocity_smoother/odometry" to="/odom"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="diff_drive_pose_controller"
        args="load yocs_diff_drive_pose_controller/DiffDrivePoseControllerNodelet mobile_base_nodelet_manager">
    <param name="spin_rate" value="25"/>
    <param name="start_enabled" value="false"/>
    <param name="base_frame_name" value="base_footprint"/>
    <param name="goal_frame_name" value="base_goal_pose"/>
    <param name="v_min" value="0.01"/>
    <param name="v_max" value="0.15"/>
    <param name="w_min" value="0.05"/>
    <param name="w_max" value="0.60"/>
    <param name="k_1" value="1.0"/>
    <param name="k_2" value="10.0"/>
    <param name="beta" value="0.2"/>
    <param name="lambda" value="2.0"/>
    <param name="dist_thres" value="0.01"/>
    <param name="orient_thres" value="0.06"/>
    <remap from="diff_drive_pose_controller/command_velocity" to="pose_controller_velocity_smoother/raw_cmd_vel"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="base_goal_pose_publisher"
        args="0.5 0.2 0.0 0.0 0.0 0.0 odom base_goal_pose 100"/>
</launch>